ship/src/main/scala/ch/epfl/bluebrain/nexus/ship/EventProcessor.scala

```diff
@@ -31,27 +30,39 @@ object EventProcessor {
 
   private val logger = Logger[EventProcessor.type]
 
-  def run(eventStream: Stream[IO, InputEvent], processors: EventProcessor[_]*): IO[Offset] = {
+  def run(eventStream: Stream[IO, InputEvent], processors: EventProcessor[_]*): IO[ImportReport] = {
     val processorsMap = processors.foldLeft(Map.empty[EntityType, EventProcessor[_]]) { (acc, processor) =>
       acc + (processor.resourceType -> processor)
     }
     eventStream
-      .evalTap { event =>
+      .evalScan(ImportReport.start) { case(report, event) =>
         processorsMap.get(event.`type`) match {
           case Some(processor) =>
-            processor.evaluate(event).onError { err =>
+            processor.evaluate(event)
+              .map { status =>
+                report + (event, status)
+              }
+              .onError { err =>
               logger.error(err)(
                 s"Error while processing event with offset '${event.ordering.value}' with processor '${event.`type`}'."
               )
             }
-          case None            => logger.warn(s"No processor is provided for '${event.`type`}', skipping...")
+          case None            =>
+            logger.warn(s"No processor is provided for '${event.`type`}', skipping...") >>
+            IO.pure(report + (event, ImportStatus.Dropped))
         }
       }
-      .scan((0, Offset.start)) { case ((count, _), event) => (count + 1, event.ordering) }
       .compile
       .lastOrError
-      .flatMap { case (count, offset) =>
-        logger.info(s"$count events were imported up to offset $offset").as(offset)
+      .flatTap { report =>
+        val header = logger.info(s"Type\tSuccess\tDropped")
+        val detailedCount = report.progress.foldLeft(header) { case (acc, (entityType, count)) =>
+          acc >> logger.info(s"$entityType\t${count.success}\t${count.dropped}")
+        }
+
+        val aggregatedCount = report.aggregatedCount
+        logger.info(s"${aggregatedCount.success} events were imported up to offset ${report.offset} (${aggregatedCount.success} have been dropped).") >>
+          detailedCount
```

Comment:
Could also stick in a `Show` instance for the report and this becomes `.flatTap(report => logger.info(report.show))`. (Keeps this cleaner and you could dump it all in a file if needed)